src/robotFactory.js

```diff
@@ -1,10 +1,76 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return (
+      `Robot: ${this.name},` +
+      `Chip version: ${this.chipVersion}, Weight: ${this.weight}`
+    );
```

Comment:
```suggestion
    return (
      `Robot: ${this.name}, ` +
      `Chip version: ${this.chipVersion}, Weight: ${this.weight}`
    );
```
one test failed because no white space here
also you can make variables for each part of string to avoid long string

```
  const robot = `Robot: ${this.name}`;
  const chip = `Chip version: ${this.chipVersion}`;
  const weight = `Weight: ${this.weight}`;

  return `${robot}, ${chip}, ${weight}`;